multi_obj_pool.h

```diff
@@ -47,17 +54,22 @@ struct ParetoFront {
 
     ParetoFront() = default;
     explicit ParetoFront(Dominator x): dominate(x) {}
+    explicit ParetoFront(Dominator x, Visitor v): dominate(x), visitor(v) {}
 
     bool add(const Obj& obj);
+    bool contains_better_than(const Obj& obj);
     template <class Predicate>
     inline void remove_if (Predicate pred) { pareto_front.remove_if(std::move(pred)); }
     inline size_t size() const { return pareto_front.size(); }
     inline const_iterator begin() const { return pareto_front.begin(); }
     inline const_iterator end() const { return pareto_front.end(); }
 
+    const Visitor& get_visitor() const { return visitor; }
```

Comment:
if my visitor have a non copyable element, and after the calculus, I what to return it.

That's pure speculation, of course.